nilearn/image/tests/test_image.py

```diff
@@ -824,3 +824,21 @@ def test_new_img_like_mgh_image():
     data = np.zeros((5, 5, 5), dtype=np.uint8)
     niimg = nibabel.freesurfer.MGHImage(dataobj=data, affine=np.eye(4))
     new_img_like(niimg, data.astype(float), niimg.affine, copy_header=True)
+
+
+def test_new_img_like_boolean_data_mgh():
+    """Check defaulting boolean input data to np.uint8 dtype is valid for
+    encoding with nibabel image class MGHImage.
+    """
+    data = np.random.randn(5, 5, 5).astype('uint8')
+    in_img = nibabel.freesurfer.MGHImage(dataobj=data, affine=np.eye(4))
+    new_img_like(in_img, data=in_img.get_fdata() > 0.5)
```

Comment:
I might be missing something but I don't understand what is tested here. I think you need to assert that the new image is composed of `uint8`, right?

Reply:
Well before `new_img_like` was failing on some image classes when boolean input was defaulting to `np.int8`. I included tests for `MGHImage` and `AnalyzeImage` because I know they would have failed but I am not sure how to test this more generally because there may be other image classes that also don't accept  `np.int8`

Reply:
I wonder if it is actually sufficient to have one test? It should generalize to all other image classes.
@effigies WDYT?

Reply:
Apologies for not getting back. I'm not sure it "generalizes" exactly, since someone could write a new format that doesn't support uint8. It seems like a good practical test, though, as these are less-featured data formats.

Reply:
Ok thanks for clarifying @effigies ! I think then we can keep it this way then unless someone else has another suggestion.